daemon/control.c

```diff
@@ -152,16 +164,42 @@ control_send_file_as_log_message_cb(const char *path, const char *file, void UNU
 			}
 		}
 		mem_free0(file_buf);
-	} else {
-		DEBUG("File %s could not be read to buffer.", str_buffer(path_str));
+
+		if (cbdata->remove_logs) {
+			bool is_current_log = false;
+			for (list_t *l = cbdata->current_logs; l; l = l->next) {
```

Comment:
unlinking sent log file should be done directly after sending the final fragment if specified by caller, reducing the implementation's complexity